llama_index/schema.py

```diff
@@ -1,14 +1,18 @@
 """Base schema for data structures."""
-from abc import abstractmethod
+from abc import abstractmethod, abstractproperty
```

Comment:
very nit: pretty sure this is deprecated. should compose abstractmethod and propert decorator to achieve this.